src/ui/public/vislib_vis_type/vislib_vis_type.js

```diff
@@ -20,6 +20,7 @@ export function VislibVisTypeVislibVisTypeProvider(Private) {
     const updateIfSet = (from, to, prop, func) => {
       if (from[prop]) {
         to[prop] = func ? func(from[prop]) : from[prop];
+        delete from[prop];
```

Comment:
Cool thanks for fixing this!

Minor, unrelated to this PR feedback - I think a comment for this updateParams function would be helpful. It was hard for me to figure out what was it was for.  :)  Plus then at some point way down the line if we stop supporting pre 5.2 we can get rid of the code.